src/cargo/ops/cargo_rustc/context.rs

```diff
@@ -192,7 +193,7 @@ impl<'a, 'cfg> Context<'a, 'cfg> {
         let mut has_cfg = true;
         let output = with_cfg.exec_with_output().or_else(|_| {
             has_cfg = false;
-            process.exec_with_output()
+            process.exec_with_output_may_fail()
```

Comment:
Hm this was added here but `status` is never checked outside? That seems like we may be papering over errors?